examples/places/components/places-example/places-example.tsx

```diff
@@ -55,7 +59,12 @@ const PlacesExample = () => {
     }
 
     service.nearbySearch(request, callback);
-  }, [map]);
+
+    // Clean up markers
+    return () => {
+      markers.forEach(marker => marker.setMap(null));
+    };
+  }, [map, Boolean(service)]);
```

Comment:
Why is it boolean here, but not at the others, e.g. `elevator` `distance-matrix-service` and `geocoder`?

Reply:
It depends on when we want the effect to rerun.
In this places examples, it doesn't matter whether there's a new places service instance or not. We already did the nearby search the first time the places service got available and don't have to do this again for a new instance. That's why I added the `Boolean()` here.
In the elevator example however, we want to update the elevator instance in the click listener each time the elevator instance changed. That's why the elevator is a dependency without `Boolean`.

I updated the distance matrix and geocoder example though.

Reply:
Okay, that's really interesting. Thanks for explaining!